Doc: Update DoConcurrentLoop documentation (#3959)
lfortran/lfortran: doc/src/asr/asr_nodes/statement_nodes/doconcurrentloop.md

### Syntax

```fortran
DoConcurrentLoop(do_loop_head head, expr* shared, expr* local, stmt* body)
```

### Arguments

`head` contains do loop concurrent header.

> `shared` contains a list of variables that are shared with all the threads
> `local` contains a list of variables that are local to each threads

Reviewer: ```suggestion
`shared` contains a list of variables that are shared amongst all threads
`local` contains a list of variables that are local to each thread
```

Minor changes, rest all is good.
